x-pack/plugins/apm/public/components/app/ErrorGroupOverview/List/index.tsx

```diff
@@ -44,15 +54,33 @@ const Culprit = styled.div`
   font-family: ${fontFamilyCode};
 `;
 
-export class ErrorGroupList extends Component {
-  state = {
+interface Props {
+  location: Location;
+  urlParams: IUrlParams;
+  items: ErrorGroupListAPIResponse;
+}
+
+interface ITableChange {
+  page: { index: number; size: number };
+  sort: {
+    field: string;
+    direction: string;
+  };
+}
+
+interface State {
+  page: { index: number; size: number };
+}
+
+export class ErrorGroupList extends Component<Props, State> {
+  public state = {
     page: {
       index: 0,
       size: 25
     }
   };
 
-  onTableChange = ({ page = {}, sort = {} }) => {
+  public onTableChange = ({ page, sort }: ITableChange) => {
```

Comment:
When EUI finishes its types, I think `sort` isn't required to show up here in this signature. In this component, we always provide `sorting` to the `EuiBasicTable` so it will always be here for us, so this works for now, but it will probably blow up once EUI pushes out the types for `EuiBasicTable`. Just noting :)

Reply:
Are you saying sort should be marked as optional?

Reply:
Hmm... but making it optional would be a breaking change. Probably misunderstanding your suggestion.

Reply:
Oh, I see what you mean. I removed the default arguments, which are required if a `sorting` prop is not set on the component.

Reply:
... and likewise with `page` if `pagination` is not provided 👍

Reply:
I think with method overloading it is possible to specify different interfaces. So if `sorting` is provided the `onTableChange` signature will have it as required. But not sure, so I'll err on the side of caution.